app/submitter/submitter.py

```diff
@@ -141,3 +143,46 @@ def send_message(self, message, tx_id, questionnaire_id, case_id=None):
         finally:
             if connection:
                 self._disconnect(connection)
+
+
+class GCSFeedback:
+    def __init__(self, bucket_name):
+        client = storage.Client()
+        self.bucket = client.get_bucket(bucket_name)
+        self.bucket.retention_period = 86400
+
+    def upload(self, schema, feedback_data, session_data):
+        object_key = str(uuid4())
+        blob = self.bucket.blob(object_key)
+        blob.metadata = {
+            "feedback_count": session_data.feedback_count + 1,
+            "form_type": schema.form_type,
+            "language_code": session_data.language_code,
+            "object_key": object_key,
```

Comment:
Not sure we need this in the payload as anytime you interact with a blob, you have access to it's name which is the key. With it being a random UUID, it doesn't provide  any additional filtering etc.

Reply:
I did think that as well, but then thought it did say it on the card